apps/src/code-studio/reporting.js

```diff
@@ -290,6 +290,12 @@ reporting.sendReport = function (report) {
           var fallback = getFallbackResponse(report) || {};
           response.redirect = fallback.redirect;
         }
+        if (appOptions.isBonusLevel) {
+          // Bonus levels might have to take students back to a different stage,
+          // ignore the redirect in response and use the url from appOptions
+          // instead
+          response.redirect = appOptions.nextLevelUrl;
```

Comment:
Where does `nextLevelUrl` come from?

Reply:
https://github.com/code-dot-org/code-dot-org/pull/19033/files#diff-1f68e16ab551449bdbd0fb7f01004f68R144

It's just always passed in appOptions now.